commit a194625ef31f0c33afae9b53e2dfaa17c2517606
Author: Samuel Thibault <samuel.thibault@ens-lyon.org>
Date:   Tue Aug 9 01:42:50 2016 +0200

    Fix recvmsg returning SIGLOST on PF_LOCAL sockets
    
    when msg_name is not NULL.
    
    	* sysdeps/mach/hurd/recvmsg.c (__libc_recvmsg): Cope with aport being
    	MACH_PORT_NULL.

diff --git a/sysdeps/mach/hurd/recvmsg.c b/sysdeps/mach/hurd/recvmsg.c
index 770a42e..51cfbd7 100644
--- a/sysdeps/mach/hurd/recvmsg.c
+++ b/sysdeps/mach/hurd/recvmsg.c
@@ -64,7 +64,7 @@ __libc_recvmsg (int fd, struct msghdr *message, int flags)
 					       &message->msg_flags, amount)))
     return __hurd_sockfail (fd, flags, err);
 
-  if (message->msg_name != NULL)
+  if (message->msg_name != NULL && aport != MACH_PORT_NULL)
     {
       char *buf = message->msg_name;
       mach_msg_type_number_t buflen = message->msg_namelen;
@@ -98,6 +98,8 @@ __libc_recvmsg (int fd, struct msghdr *message, int flags)
       if (buflen > 0)
 	((struct sockaddr *) message->msg_name)->sa_family = type;
     }
+  else if (message->msg_name != NULL)
+    message->msg_namelen = 0;
 
   __mach_port_deallocate (__mach_task_self (), aport);
 
